doc/development/plugins.rst

```diff
@@ -1,397 +1,398 @@
-Furthermore, PennyLane uses the convention :math:`|q_0,q_1,\dots,q_{N-1}\rangle` where
-:math:`q_0` is the most significant bit.
+1) Strictly use wires provided by the user on initialization ``device(name, wires=wires)``
+2) Infer the number and ordering of wires provided by the submitted circuit.
+3) Strictly require specific wire labels
 
-And thats it! The device has inherited :meth:`~.QubitDevice.expval`, :meth:`~.QubitDevice.var`,
-and :meth:`~.QubitDevice.sample` methods, that accepts an observable (or tensor product of
-observables) and returns the corresponding measurement statistic.
+Option 2 allows workflows to change the number and labeling of wires over time, but sometimes users want
+to enfore a wire convention and labels. If a user does provide wires, the :meth:`~.devices.Device.preprocess` should
+validate that submitted circuits only have wires in the requested range.
 
+>>> dev = qml.device('default.qubit', wires=1)
+>>> circuit = qml.tape.QuantumScript([qml.CNOT((0,1))], [qml.state()])
+>>> dev.preprocess()[0]((circuit, ))
+WireError: Cannot run circuit(s) of default.qubit as they contain wires not found on the device.
 
+PennyLane wires can be any hashable object, where wire labels are distinguished by their equality and hash.
+If working with successive integers (``0``, ``1``, ``2``, ...) is preferred internally,
+the :meth:`~.QuantumScript.map_to_standard_wires` method can be used inside of 
+the :meth:`~.devices.Device.execute` method.  The :class:`~.map_wires` transform can also 
+map the wires of the submitted circuit to internal labels.
 
-:html:`<div class="caution admonition" id="aside1"><a data-toggle="collapse" data-parent="#aside1" href="#content1" class="collapsed"><p class="first admonition-title">Advanced execution control (click to expand) <i class="fas fa-chevron-circle-down"></i></p></a><div id="content1" class="collapse" data-parent="#aside1" style="height: 0px;">`
+Sometimes hardware qubit labels cannot be arbitrarily mapped without a changing in behaviour.
+Connectivity, noise, performance, and
+other constraints can make it so that operations on qubit one cannot be arbitrarily exchanged with the same operation
+of qubit two.  In such a situation, the device can hard code a list of the only acceptable wire labels. In such a case, it
+will be on the user to deliberately map wires if they wish such a thing to occur.
 
-Additional flexibility is sometimes required for interfacing with more
-complicated frameworks.
+>>> qml.device('my_hardware').wires
+<Wires = [0, 1, 2, 3]>
+>>> qml.device('my_hardware', wires=(10, 11, 12, 13))
+DeviceError: Device my_hardware cannot internally map wires, as the labels 0, 1, 2, 3 indicate unique qubits
 
-When PennyLane needs to evaluate a QNode, it accesses the :meth:`~.QubitDevice.execute` method of
-your plugin which, by default, performs the following process:
+To implement such validation, a device developer can include a custom transform in their preprocessing:
```

Comment:
Why not simply use the `validate_device_wires` provided by PL?

Reply:
different error message. 

Though actually such an error message should be handled on initialization.  Let me fix that.